### PR TITLE
Sprint 2/Login, Registration, and Main Pages Implementation

### PR DESCRIPTION
**Pull request by ViktoriyaDatchuk**
1. Task [Sprint 2: Login, Registration, and Main Pages Implementation](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint%232.md)
2. Screenshot: 

3. Deploy: [kinogo-vno](https://kinogo-vno.netlify.app/)
4. Done: 20.05.2024 / deadline 21.05.2024
5. Score: 315 / 315
Login Page Implementation (130 points - Total):
Input Validation (40 points)
Integration with Authentication Service (45 points)
Redirection (30 points)
Handle Authentication Token (10 points)
Navigation to Registration Page (5 points)
Registration Page Implementation (120 points - Total):
Input Validation (45 points)
Integration with Authentication Service (25 points)
State Management, Automatic Login, and Redirection (15 points)
Integration with commercetools for User Profiles and Addresses (30 points)
Navigation to Login Page (5 points)
Main Page Enhancements (10 points - Total):
Centralized Navigation (10 points)
Routing Implementation (30 points - Total)
Evaluation Criteria for Header (25 points - Total)

### CODE REVIEW

**Review by Jezhora — APPROVED** [review]

File: eCommerce/src/pages/SignUp.tsx [hunk 1]
```diff
@@ -33,11 +33,15 @@ enum ErrorMessagesReg {
...
+  const validStyle = `${inputStyle} border-green-500`;
+  const validStyleShort = `${inputStyleShort} border-green-500`;
+  const invalidStyle = `${inputStyle} text-slate-950 text-red-400 border-red-500`;
+  const invalidStyleShort = `${inputStyleShort} text-slate-950 text-red-400 border-red-500`;
+  const labelStyleRegular = 'text-orange-400 text-lg text-left flex flex-col mt-2';
   const labelStyleShort = 'text-orange-400 text-lg text-left flex flex-col';
+  const labelStyleCheckbox = 'text-teal-400 flex flex-row mt-4';
   const wrapperStyle = 'flex flex-row flex-wrap gap-x-8 mt-4 gap-y-4';
   const labelStyleGreen = 'text-teal-400 text-lg text-left flex flex-col';
   const errorStyle = 'text-red-400 text-sm';
```
Comment (drobov1k) [inline eCommerce/src/pages/SignUp.tsx:47]: Could all these constants be in a separate file (style.js)?
When I read a component, I want to focus on the logic.

File: eCommerce/src/user/userAuth.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
+// import { Address, ClientResponse, MyCustomerUpdateAction } from '@commercetools/platform-sdk';
+import { Address, ClientResponse } from '@commercetools/platform-sdk';
+import apiRoot from '../sdk/apiRoot';
+import { FormSignIn } from '../components/SignInForm';
+import { IFormInput } from '../pages/SignUp';
+
+const saveUserToLocalStorage = (response: ClientResponse, navigate: (path: string) => void) => {
```
Comment (drobov1k) [inline eCommerce/src/user/userAuth.ts:7]: Why should this utility function know about a response object? `saveUserToLocalStorage` should accept `User` or `Partial<User>`.

File: eCommerce/src/pages/SignUp.tsx [hunk 3]
```diff
@@ -46,208 +50,241 @@ export default function SignUp() {
...
+            <label htmlFor="email" className={labelStyleRegular}>
+              Email
+              <input
+                type="email"
+                {...register('email', {
+                  required: ErrorMessagesReg.REQUIRED,
+                  pattern: {
+                    value:
+                      /* eslint-disable no-useless-escape */
+                      /^(([^<>()[\]\.,;:\s@\"]+(\.[^<>()[\]\.,;:\s@\"]+)*)|(\".+\"))@(([^<>()[\]\.,;:\s@\"]+\.)+[^<>()[\]\.,;:\s@\"]{2,})$/i,
```
Comment (drobov1k) [inline eCommerce/src/pages/SignUp.tsx:86]: I'd suggest to extract into constant

File: eCommerce/src/pages/SignUp.tsx [hunk 4]
```diff
@@ -46,208 +50,241 @@ export default function SignUp() {
...
+          </form>
         </div>
-        <p className="text-orange-400 mt-8">
-          Already have an account?{' '}
-          <button type="button" onClick={() => navigate('/sign-in')} className="underline">
-            sign-in here
-          </button>
-        </p>
-      </form>
+      </div>
```
Comment (drobov1k) [inline eCommerce/src/pages/SignUp.tsx:287]: This component is getting enormous. Need to think about splitting it.

File: eCommerce/src/pages/SignUp.tsx [hunk 5]
```diff
@@ -46,208 +50,241 @@ export default function SignUp() {
...
 
   const navigate = useNavigate();
 
-  const onSubmit: SubmitHandler<IFormInput> = (data) => {
-    sighUpUser(data, navigate);
-  };
-
   const [postCodeRegExp, setPostCodeRegExp] = useState('^\\d{4}$');
   const [postCodeFormat, setPostCodeFormat] = useState('NNNN');
+  const [isDefaultAddress, setIsDefaultAddress] = useState(false);
```
Comment (drobov1k) [inline eCommerce/src/pages/SignUp.tsx:59]: I don't understand why you use state for this purpose if you're using a form library.

File: eCommerce/src/components/Header/Profil/LogIn.tsx [hunk 6]
```diff
@@ -11,13 +11,12 @@ export default function LogIn() {
 
   const userString = localStorage.getItem('commercetools_user');
   const user = userString ? JSON.parse(userString) : null;
```
Comment (drobov1k) [inline eCommerce/src/components/Header/Profil/LogIn.tsx:13]: It's time to introduce `useUser` hook?

File: eCommerce/src/interfaces/userData.interface.ts [hunk 7]
```diff
@@ -0,0 +1,10 @@
+export interface Address {
+  key: string;
+  firstName: string;
+  lastName: string;
```
Comment (drobov1k) [inline eCommerce/src/interfaces/userData.interface.ts:4]: Are these fields related to `Address`?

File: eCommerce/src/user/userAuth.ts [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+        defaultBillingAddress,
+        defaultShippingAddress,
+      },
+    })
+    .execute()
+    .then((response) => {
+      saveUserToLocalStorage(response, navigate);
+    })
+    .catch((errors: Error) => {
+      if (errors.message === 'There is already an existing customer with the provided email.') {
```
Comment (drobov1k) [inline eCommerce/src/user/userAuth.ts:122]: Maybe we can catch by code?

File: eCommerce/src/user/userAuth.ts [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+      city: userData.city,
+      country: userData.country,
+      email: userData.email,
+    });
+  }
+  return addresses;
+};
+
+export function sighUpUser(
+  userData: IFormInput,
```
Comment (drobov1k) [inline eCommerce/src/user/userAuth.ts:84]: I'd suggest to pass an object
